File: Caching.fs


open System.Linq

let mutable k = 0
let aeiou =
    k <- k + 1
    let ls = ["a"; "e"; "e"; "i"; "o"; "u"]
    fun input -> ls.Contains(input)

aeiou "a"
aeiou "c"
k
k
aeiou "c"
k
aeiou "u"
k
k

File: Caching.js

var k = 0;

function aeiou() {
  k = k + 1
  var ls = ["a", "e", "i", "o", "u"];
  return function(input) {
    return ls.indexOf(input) != -1;
  };
}

var f = aeiou()
f("a")
k
k
f("b")
k

k
aeiou()("a")
k
aeiou()("i")
k